stdlib/os/__init__.pyi

```diff
@@ -349,13 +349,17 @@ class stat_result(structseq[float], tuple[int, int, int, int, int, int, int, flo
         @property
         def st_rdev(self) -> int: ...  # type of device if an inode device
         if sys.platform != "linux":
-            # These properties are available on MacOS, but not on Windows or Ubuntu.
             # On other Unix systems (such as FreeBSD), the following attributes may be
             # available (but may be only filled out if root tries to use them):
             @property
             def st_gen(self) -> int: ...  # file generation number
-            @property
-            def st_birthtime(self) -> int: ...  # time of file creation
+
+    if (sys.version_info >= (3, 12) and sys.platform == "win32") or (sys.platform != "linux" and sys.platform != "win32"):
+        @property
+        def st_birthtime(self) -> int: ...  # time of file creation in seconds
```

Comment:
I think this can be improved, open to suggestions:
- Either repeat the definition of`st_birthtime` for win32 and non-linux platforms
- Or add some nested condition to for this